stratum/hal/lib/tdi/tofino/tofino_sde_wrapper.cc

```diff
@@ -90,17 +111,27 @@ std::string GetBfChipId(int device) {
 
 }  // namespace
 
-std::string TdiSdeWrapper::GetChipType(int device) const {
+::util::StatusOr<bool> TofinoSdeWrapper::IsSoftwareModel(int device) {
+  bool is_sw_model = true;
+  auto bf_status = bf_pal_pltfm_type_get(device, &is_sw_model);
+  RET_CHECK(bf_status == BF_SUCCESS) << "Error getting software model status.";
+  return is_sw_model;
+}
+
+std::string TofinoSdeWrapper::GetChipType(int device) const {
   return absl::StrCat(GetBfChipFamilyAndType(device), ", revision ",
                       GetBfChipRevision(device), ", chip_id ",
                       GetBfChipId(device));
 }
 
-std::string TdiSdeWrapper::GetSdeVersion() const { return "9.11.0"; }
+std::string TofinoSdeWrapper::GetSdeVersion() const { return "9.11.0"; }
```

Comment:
Hardcode to 9.11.0 or is this also a `TODO` workitem?

Reply:
This is the result of paring down the equivalent function in BfSdeWrapper:
```c
std::string BfSdeWrapper::GetSdeVersion() const {
#if defined(SDE_9_1_0)
  return "9.1.0";
#elif defined(SDE_9_2_0)
  return "9.2.0";
#elif defined(SDE_9_3_0)
  return "9.3.0";
#elif defined(SDE_9_3_1)
  return "9.3.1";
#elif defined(SDE_9_3_2)
  return "9.3.2";
#elif defined(SDE_9_4_0)
  return "9.4.0";
#elif defined(SDE_9_5_0)
  return "9.5.0";
#else
#error Unsupported SDE version
#endif
}
```
The version of the Tofino SDE that supported the IPU was 9.11.0; hence the hard-coded value.
There's been no new Tofino development since then.